doc/developer-guides/memmgt-hld.rst

```diff
@@ -0,0 +1,249 @@
+platform E820 table. The following sections demonstrate different memory
+space management by referencing it.
+
+Physical to Virtual Mapping
+===========================
+
+ACRN hypervisor is running under paging mode, so after receiving
+the platform E820 table, ACRN hypervisor creates its MMU page table
+based on it. This is done by the function init_paging() for all
+physical CPUs.
+
+The memory mapping policy here is:
+
+-  Identical mapping for each physical CPU (ACRN hypervisor's memory
+   could be relocatable in a future implementation)
+-  Map all memory regions with UNCACHED type
+-  Remap RAM regions to WRITE-BACK type
+
+.. figure:: images/mem-image4.png
+   :align: center
+   :width: 900px
+   :name: vm-layout
+
+   Hypervisor Virtual Memory Layout
+
+:numref:`vm-layout` shows:
+
+-  Hypervisor can access all of system memory
+-  Hypervisor has an UNCACHED MMIO/PCI hole reserved for devices, such
+   as for LAPIC/IOAPIC accessing
+-  Hypervisor has its own memory with WRITE-BACK cache type for its
+   code and data (< 1M part is for secondary CPU reset code)
+
+Service OS Memory Management
+****************************
+
+After the ACRN hypervisor starts, it creates the Service OS as its first
+VM. The Service OS runs all the native device drivers, manage the
+hardware devices, and provides I/O mediation to guest VMs. The Service
+OS is in charge of the memory allocation for Guest VMs as well.
+
+ACRN hypervisor passes the whole system memory access (except its own
+part) to the Service OS. The Service OS must be able to access all of
+the system memory except the hypervisor part.
+
+Guest Physical Memory Layout - E820
+===================================
+
+The ACRN hypervisor passes the original E820 table to the Service OS
+after filtering out its own part. So from Service OS's view, it sees
+almost all the system memory as shown here:
+
+.. figure:: images/mem-image3.png
+   :align: center
+   :width: 900px
+   :name: sos-mem-layout
+
+   SOS Physical Memory Layout
+
+Host to Guest Mapping
+=====================
+
+ACRN hypervisor creates Service OS's host (HPA) to guest (GPA) mapping
+(EPT mapping) through the function
+prepare_vm0_memmap_and_e820() when it creates SOS VM. It follows
+these rules:
+
+-  Identical mapping
+-  Map all memory range with UNCACHED type
+-  Remap RAM entries in E820 (revised) with WRITE-BACK type
+-  Unmap ACRN hypervisor memory range
+-  Unmap ACRN hypervisor emulated vLAPIC/vIOAPIC MMIO range
+
+The host to guest mapping is static for the Service OS; it will not
+change after the Service OS begins running. Each native device driver
+can access its MMIO through this static mapping. EPT violation is only
+serving for vLAPIC/vIOAPIC's emulation in the hypervisor for Service OS
+VM.
+
+User OS Memory Management
+*************************
+
+User OS VM is created by the DM (Device
+eMulator) application running in the Service
```

Comment:
we usually refer to DM as Device Model, we should stick to that unless there is a specific reason not to.